backend/dal/Helpers/Extensions/IdentityExtensions.cs

```diff
@@ -120,6 +121,38 @@ public static bool HasRoles(this ClaimsPrincipal user, params string[] role)
             return count == role.Length;
         }
 
+        /// <summary>
+        /// Determine if the user any of the specified roles.
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="role"></param>
+        /// <returns>True if the user has any of the roles.</returns>
+        public static bool HasRole(this ClaimsPrincipal user, params RoleClaim[] role)
+        {
+            if (role == null) throw new ArgumentNullException(nameof(role));
+            if (role.Length == 0) throw new ArgumentOutOfRangeException(nameof(role));
+
+            var roles = role.Select(r => r.GetRoleName()).ToArray();
+            return user.Claims.Any(c => c.Type == ClaimTypes.Role && roles.Contains(c.Value));
+        }
+
+        /// <summary>
+        /// Determine if the user all of the specified roles.
+        /// </summary>
+        /// <param name="user"></param>
+        /// <param name="role"></param>
+        /// <returns>True if the user has all of the roles.</returns>
+        public static bool HasRoles(this ClaimsPrincipal user, params RoleClaim[] role)
+        {
+            if (role == null) throw new ArgumentNullException(nameof(role));
+            if (role.Length == 0) throw new ArgumentOutOfRangeException(nameof(role));
+
+            var roles = role.Select(r => r.GetRoleName()).ToArray();
+            var count = user.Claims.Count(c => c.Type == ClaimTypes.Role && roles.Contains(c.Value));
```

Comment:
Is there any benefit to this Count() method over just using All()?

Reply:
All() would be a nicer way of writing it.